### PR TITLE
[ENG-1196] Allow `WEBUI_PORT` to be specified for host-networked containers

### PR DESCRIPTION
**Pull request by brxken128**
This also changes the Dockerfile to follow some more of the best practices (it's missing a few but it's a lot better, although I may have broke something along the way).

I also cleaned up the other CI files, as `trunk` was emitting quite a few lint warnings - I'm not sure what effects my changes will have, we will have to see.

**Please don't merge or approve this yet, it's experimental but not a draft so CI still runs**

### CODE REVIEW

File: apps/server/docker/entrypoint.sh [hunk 1]
```diff
@@ -26,11 +26,11 @@ adduser --system --disabled-password \
   spacedrive
 passwd -l spacedrive
 
-if [ -n "${TZ:-}" ]; then
```
Comment (HeavenVolkoff) [inline apps/server/docker/entrypoint.sh:29]: The colon here is required, as it means substitute when unset or null, without a color it only substitutes when unset.
https://pubs.opengroup.org/onlinepubs/9699919799/utilities/V3_chap02.html#tag_18_06_02
image

File: apps/server/docker/entrypoint.sh [hunk 2]
```diff
@@ -5,9 +5,9 @@ set -eu
 # Shortcircuit for non-default commands.
 # The last part inside the "{}" is a workaround for the following bug in ash/dash:
 # https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=874264
-if [ -n "${1:-}" ] && [ "${1#-}" = "${1}" ] \
```
Comment (HeavenVolkoff) [inline apps/server/docker/entrypoint.sh:8]: The colon here is required, as it means substitute when unset or null, without a color it only substitutes when unset.

**Review by HeavenVolkoff — CHANGES_REQUESTED** [review]

File: apps/server/docker/Dockerfile [hunk 3]
```diff
@@ -60,9 +70,11 @@ RUN pnpm web build
 
 FROM build-base as server
 
-RUN wget -qO- https://sh.rustup.rs | sh -s -- -yq --profile minimal
+# trunk-ignore(hadolint/DL4006)
+RUN ["/bin/bash", "-c", "set -o pipefail && wget -qO- https://sh.rustup.rs | sh -s -- -yq --profile minimal"]
```
[the remark this answers is not in the document]
Comment (brxken128) [reply apps/server/docker/Dockerfile:74]: https://docs.docker.com/develop/develop-images/instructions/#using-pipes

File: apps/server/docker/Dockerfile [hunk 4]
```diff
@@ -8,25 +8,35 @@ FROM debian:bookworm as base
 
 ARG DEBIAN_FRONTEND
 
-ADD https://gist.githubusercontent.com/HeavenVolkoff/ff7b77b9087f956b8df944772e93c071/raw \
+COPY https://gist.githubusercontent.com/HeavenVolkoff/ff7b77b9087f956b8df944772e93c071/raw \
 	/etc/apt/apt.conf.d/99docker-apt-config
 
+USER spacedrive
+HEALTHCHECK NONE
```
[the remark this answers is not in the document]
Comment (brxken128) [reply apps/server/docker/Dockerfile:15]: Best practices recommend at least defining a `HEALTHCHECK` per layer.

File: apps/server/docker/entrypoint.sh (hunk 1 above)
Comment (brxken128) [reply apps/server/docker/entrypoint.sh:29]: I don't recall touching this line, maybe auto-formatting

**Comment by brxken128** [comment]
Just going to close this, all I was trying to do was allow exposing the WEBUI on a different port for people who use host networking so their container can see other nodes wtithout having to worry about ports, and so that 8080 doesn't clash with another container (which will almost 100% happen).

Was very clearly a WIP as I'd mentioned at the start, only reason it wasn't a draft is because I wanted CI to keep running.

**Comment by HeavenVolkoff** [comment]
> Just going to close this, all I was trying to do was allow exposing the WEBUI on a different port for people who use host networking so their container can see other nodes wtithout having to worry about ports, and so that 8080 doesn't clash with another container (which will almost 100% happen).

I think this is a valid addition, as almost all images have a similar option exposed. Could we re-open this?

**Pull request closed without merging**